driver-core/src/main/com/mongodb/client/model/expressions/DocumentExpression.java

```diff
@@ -43,6 +45,7 @@ default BooleanExpression getBoolean(final String fieldName, final boolean other
         return getBoolean(fieldName, of(other));
     }
 
+    @MqlUnchecked(TYPE)
     NumberExpression getNumber(String fieldName);
```

Comment:
I wonder if it should be called MqlAsserted(TYPE) instead? It seems to read a bit more easily, and suggest something on the user's part, rather than something perhaps unavoidable in the API? Unsure.

Reply:
While the Java SE API does not provide conceptually similar methods, the Java language has a concept of _unchecked_ conversions. The Rust standard library provides functions where a user is supposed to guarantee something that is not checked by the function; it uses the [_"unchecked"_ suffix in the names of those functions](https://doc.rust-lang.org/std/?search=_unchecked), and even has an explicit [corresponding naming convention](https://rust-lang.github.io/api-guidelines/dependability.html?highlight=unchecked#dynamic-enforcement:~:text=The%20convention%20is%20to%20mark%20these%20opt%2Dout%20functions%20with%20a%20suffix%20like%20_unchecked%20or%20by%20placing%20them%20in%20a%20raw%20submodule.).

That's why I think that "unchecked" is a better word - we are following existing conventions.